queries/djot/highlights.scm

```diff
@@ -0,0 +1,353 @@
+(code_block
+  .
+  (code_block_marker_begin)
+  (language)
+  (code) @none)
+
+[
+  (code_block_marker_begin)
+  (code_block_marker_end)
+  (raw_block_marker_begin)
+  (raw_block_marker_end)
+] @punctuation.delimiter
+
+(language) @tag.attribute
+
+(language_marker) @tag.delimiter
+
+(class_name) @tag.attribute
+
+(block_quote) @markup.quote
+
+(block_quote_marker) @punctuation.special
+
+(table_header) @text.title
+
+(table_header
+  "|" @punctuation.special)
+
+(table_row
+  "|" @punctuation.special)
+
+(table_separator) @punctuation.special
+
+(table_caption
+  (marker) @punctuation.special)
+
+(table_caption) @markup.caption
+
+[
+  (list_marker_dash)
+  (list_marker_plus)
+  (list_marker_star)
+  (list_marker_definition)
+  (list_marker_decimal_period)
+  (list_marker_decimal_paren)
+  (list_marker_decimal_parens)
+  (list_marker_lower_alpha_period)
+  (list_marker_lower_alpha_paren)
+  (list_marker_lower_alpha_parens)
+  (list_marker_upper_alpha_period)
+  (list_marker_upper_alpha_paren)
+  (list_marker_upper_alpha_parens)
+  (list_marker_lower_roman_period)
+  (list_marker_lower_roman_paren)
+  (list_marker_lower_roman_parens)
+  (list_marker_upper_roman_period)
+  (list_marker_upper_roman_paren)
+  (list_marker_upper_roman_parens)
+] @markup.list
+
+(list_marker_task
+  (unchecked) @constant.builtin) @markup.list.unchecked
+
+(list_marker_task
+  (checked) @constant.builtin) @markup.list.checked
+
+(list_item
+  (term) @define)
+
+[
+  (ellipsis)
+  (en_dash)
+  (em_dash)
+  (straight_quote)
+] @string.special
+
+(frontmatter_marker) @punctuation.delimiter
+
+(emphasis) @markup.italic
+
+(strong) @markup.strong
+
+(emphasis
+  (emphasis_begin) @punctuation.delimiter)
+
+(emphasis
+  (emphasis_end) @punctuation.delimiter)
+
+(strong
+  (strong_begin) @punctuation.delimiter)
+
+(strong
+  (strong_end) @punctuation.delimiter)
+
+(highlighted) @markup.highlighted
+
+(highlighted
+  [
+    "{="
+    "=}"
+  ] @punctuation.delimiter)
+
+(insert) @markup.insert
+
+(insert
+  [
+    "{+"
+    "+}"
+  ] @punctuation.delimiter)
+
+(delete) @markup.delete
+
+(delete
+  [
+    "{-"
+    "-}"
+  ] @punctuation.delimiter)
+
+(symbol) @markup.symbol
+
+(superscript) @markup.superscript
+
+(superscript
+  [
+    "^"
+    "{^"
+    "^}"
+  ] @punctuation.delimiter)
+
+(subscript) @markup.subscript
+
+(subscript
+  [
+    "~"
+    "{~"
+    "~}"
+  ] @punctuation.delimiter)
+
+(verbatim) @markup.raw
+
+[
+  (verbatim_marker_begin)
+  (verbatim_marker_end)
+] @punctuation.delimiter
+
+(math) @markup.math
+
+[
+  (math_marker)
+  (math_marker_begin)
+  (math_marker_end)
+] @punctuation.delimiter
+
+(raw_inline) @markup.raw
+
+[
+  (raw_inline_attribute)
+  (raw_inline_marker_begin)
+  (raw_inline_marker_end)
+] @punctuation.delimiter
+
+(paragraph) @spell
+
+(span
+  [
+    "["
+    "]"
+  ] @punctuation.bracket)
+
+(inline_attribute
+  [
+    "{"
+    "}"
+  ] @punctuation.bracket)
+
+(block_attribute
+  [
+    "{"
+    "}"
+  ] @punctuation.bracket)
```

Comment:
I see a lot of this bracket. Probably can just narrow it all to just 
```suggestion
[
  "{"
  "}"   
  "["
  "]"
  "("
  ")"
] @punctuation.bracket
```

Reply:
They're nested under specific nodes so that a random `[` doesn't get highlighted like a bracket, only those that are in the right spot. Is that not a goal of nvim-treesitter's queries?

Reply:
Random `[` never get captured, only (named or anonymous) nodes. So it depends whether other rules in the grammar contain a literal `[` or not.

Reply:
There are rules such as `{+`, which is different than `{`. There are others I think, too.

Reply:
{+ literal node is still not { literal node, so my point still stands

Reply:
Done.